app/v4/models/request/submitFinalDeclaration.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v4.models.request
+
+import api.models.domain.{CalculationId, Nino, TaxYear}
+import api.models.request.RawData
+import play.api.libs.json.{JsValue, Json}
+
+case class SubmitFinalDeclarationRawData(nino: String, taxYear: String, calculationId: String) extends RawData
+
+case class SubmitFinalDeclarationRequest(nino: Nino, taxYear: TaxYear, calculationId: CalculationId) {
+  def toNrsJson: JsValue = Json.obj("calculationId" -> calculationId.calculationId)
```

Comment:
We should change this to:
```suggestion
  def toNrsJson: JsValue = Json.obj("calculationId" -> s"calculationId")
```

Reply:
(no need, same as for v3)